### ISSUE TITLE
Routing and Redirection

### ISSUE THREAD

**Issue by ajsyp** [David-OConnor/seed#39]
I am trying to implement some simple login redirection for my web frontend using Seed: if the user navigates to a protected page but is not logged in, redirect to the login page (by pushing a route).

This feels like logic that ought to be in my model's `update` call. However, `seed::push_route` requires the application `App<Msg, Model>` object. Meanwhile, all the code examples seem to be capturing the `App` in the `view()` method, and using an event to trigger the route-pushing. How would I trigger this event programmatically? All the events (e.g., `simple_ev(...)`) seem to be related to user interaction, but I want my application to do this automatically. Some sort of "send event to myself" call would seem in order (e.g., a hypothetical `seed::queue_message(...)`).

I have additionally tried calling `seed::window().location().assign(...)`, but this seems to lead to redirections loops--the browser refreshes/redirects itself over and over to the same page.

Any ideas on how to inspect the model and programmatically push a route?

**Comment by David-OConnor** [comment]
Does something like this work?
```rust
fn view(state: seed::App<Msg, Model>, model: &Model) -> El<Msg> {  
    div![
        did_mount(move |_| {
            if !model.logged_in {
                state.update(Msg::SetPage(Page::Login))
            } 
    ]
}
```

**Comment by David-OConnor** [comment]
On second thought, the example I posted is probably invalid; not sure if there's a way to do this currently. Re the new routing system, not sure. Hopefully in the next week.

**Comment by David-OConnor** [comment]
In the latest commit, this works, using code like I posted above with did_mount.

**Comment by David-OConnor** [comment]
Published in v0.2.5

**Comment by ajsyp** [comment]
Can confirm that this seems to work, thanks!
